### PR TITLE
CCTRI 1344/add refer observables endpoint logic

### PR DESCRIPTION
**Pull request by GeorgiyII**

### CODE REVIEW

File: api/enrich.py [hunk 1]
```diff
@@ -312,6 +362,29 @@ def observe_observables():
...
+    relay_input = get_json(ObservableSchema(many=True))
+
+    observables = group_observables(relay_input)
+
+    if not observables:
+        return jsonify_data({})
+
+    api_key = get_jwt().get('key', '')
+
+    client = URLScanClient(
```
Comment (gdavoian) [inline api/enrich.py:374]: Do you really need a client to build references for an observable? This endpoint doesn't even do any requests to the external service.

File: api/enrich.py [hunk 2]
```diff
@@ -312,6 +362,29 @@ def observe_observables():
...
+    g.references = []
+
+    for observable in observables:
+
+        output = client.get_search_data(observable)
+
+        if output and output['results']:
+            g.references.extend(extract_references(observable))
+
+    return jsonify_data(g.references) or jsonify_data({[]})
```
Comment (gdavoian) [inline api/enrich.py:390]: `jsonify_data` returns a Flask Response object, so `jsonify_data(g.references)` will always evaluate to `True` in the boolean context, so `jsonify_data({[]})` is useless.

File: api/enrich.py (hunk 1 above)
Comment (GeorgiyII) [reply api/enrich.py:374]: I think this is not needed too, but in the Use Case Michael wrote: /refer/observables options query the API first to make sure there are results before presenting the Browse option.

File: api/enrich.py (hunk 2 above)
Comment (GeorgiyII) [reply api/enrich.py:390]: done

File: api/enrich.py (hunk 1 above)
Comment (gdavoian) [reply api/enrich.py:374]: Hm, interesting. Probably this is the first integration with such behavior.

**Pull request merged**